### PR TITLE
PT-876 Flexible trends

### PR DESCRIPTION
**Pull request by yvon-dblg**
For the reviewer:
- I have incorporate the code of "react-transition-group-plus" directly into the viz code, with some fix. This is somewhat a out of scope change. The reason is that this component is too old, and create lots of messages in the console. I needed it to be silent, in order to have less annoying messages.
- There is some change in the React flow (like componentWillUpdate -> componentDidUpdate); Theses was deprecated in React 15. They are here to be able to migrate to React 16 one day (and avoid warning message in the console too).
If theses changes are unacceptable, I can remove them ;-)

### CODE REVIEW

**Comment by franck-fourel** [comment]
@yvon-dblg  Not sure to understand why we do not use the npm module, react-transition-group-plus. 
If we need to fix it for any good reasons, that would make sense to fork it and publish the npm module in our internal nexus instead of having the code of the module within viz.

**Comment by yvon-dblg** [comment]
> 
> 
> @yvon-dblg Not sure to understand why we do not use the npm module, react-transition-group-plus.
> If we need to fix it for any good reasons, that would make sense to fork it and publish the npm module in our internal nexus instead of having the code of the module within viz.

I should probably remove this code, and keep it just for myself.
The reason was that this plugin is obsolete for React 15. It display tons of warning messages, which pollute the console. when you debug something you prefer to have only stuff you're interested in.

**Comment by franck-fourel** [comment]
> > @yvon-dblg Not sure to understand why we do not use the npm module, react-transition-group-plus.
> > If we need to fix it for any good reasons, that would make sense to fork it and publish the npm module in our internal nexus instead of having the code of the module within viz.
> 
> I should probably remove this code, and keep it just for myself.
> The reason was that this plugin is obsolete for React 15. It display tons of warning messages, which pollute the console. when you debug something you prefer to have only stuff you're interested in.

yep I got it. 
would that make sense to fork it and register it as a module in our nexus?

File: src/components/trends/common/TrendsContainer.js [hunk 1]
```diff
@@ -461,18 +468,25 @@ export class TrendsContainer extends PureComponent {
...
         end = prevEnd;
-      } else if (prevStart < currentStart) {
+      } else if (prevStart < dateDomain.start) {
         start = prevStart;
       }
     }
+
     return (
       <TrendsSVGContainer
         activeDays={this.props.activeDays}
```
Comment (franck-fourel) [inline src/components/trends/common/TrendsContainer.js]: Correct me if I'm wrong but most of those changes here are purely a different way to write the same thing. 
Was it really necessary ?

**Review by franck-fourel — APPROVED** [review]

**Comment by franck-fourel** [comment]
@yvon-dblg can you please change the Changelog and package.json with v.0.10.0 so that I can merge it and publish it with correct version number

File: src/components/trends/common/TrendsContainer.js (hunk 1 above)
Comment (yvon-dblg) [reply src/components/trends/common/TrendsContainer.js]: True.
I rewrite 2 or 3 times this piece of code due to some crash. I didn't bother put it back closest to the original one.

**Comment by yvon-dblg** [comment]
> yep I got it.
> would that make sense to fork it and register it as a module in our nexus?

Maybe, but for just 385 lines of code, it seems lots of trouble.
Should we ask others team members ?

**Review by franck-fourel — APPROVED** [review]
LGTM

**Pull request merged**